alerts/storage_alerts.libsonnet

```diff
@@ -23,7 +23,7 @@
           {
             alert: 'KubePersistentVolumeFullInFourDays',
             expr: |||
-              predict_linear(kubelet_volume_stats_available_bytes{%(kubeletSelector)s}[1h], 4 * 24 * 3600) < 0
+              kubelet_volume_stats_available_bytes{%(kubeletSelector)s} and predict_linear(kubelet_volume_stats_available_bytes{%(kubeletSelector)s}[%(volumeFullPredictionSampleTime)s], 4 * 24 * 3600) < 0
```

Comment:
I think you actually meant to reverse the two sides of this `and` statement. This way, the resulting numbers will be whatever is in the `kubelet_volume_stats_available_bytes`. I think you wanted to do:

```
predict_linear(kubelet_volume_stats_available_bytes{%(kubeletSelector)s}[%(volumeFullPredictionSampleTime)s], 4 * 24 * 3600) and kubelet_volume_stats_available_bytes{%(kubeletSelector)s} < 0
```

The respective PromQL documentation:

> vector1 and vector2 results in a vector consisting of the elements of vector1 for which there are elements in vector2 with exactly matching label sets. Other elements are dropped. The metric name and values are carried over from the left-hand side vector.

Reply:
Yeah, the reason behind this order was to decrease the load of the query as the prediction would not be executed when the volume does not exist anymore. I'm also not sure if it's interesting how far below 0 the prediction is (might be though), maybe a more general approach would be to make this configurable

Reply:
Makes sense, in that case we should use the value in the message. Maybe the new message should be:

```
Based on recent sampling, the persistent volume claimed by {{ $labels.persistentvolumeclaim }} in namespace {{ $labels.namespace }} is expected to fill up within four days, currently at {{ $value }}.
```

Reply:
fixed